### PR TITLE
Added start of "release_assert()"

### LINKED ISSUE #1098: Create a release-assert mechanism
Currently we use `assert(3)` for some checks that should be done in release builds, but `assert(3)` is a no-op in release builds.

Let's create an assert-alike that aborts if those checks fail in release builds.

Possible names:
 - `always_assert()`
 - `assert()` (and rename `assert(3)` to `debug_assert()`, but this is non-standard and confusing to developers)
 - `release_assert()`

### PR DESCRIPTION
**Pull request by rkeene**
Closes #1098

### CODE REVIEW

File: rai/lib/utility.cpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+void release_assert_internal (bool check, const char * check_expr, const char * file, unsigned int line)
+{
+	if (check)
+	{
+		return;
+	}
+
+	std::cerr << "Assertion (" << check_expr << ") failed " << file << ":" << line << std::endl;
+	abort();
```
Comment (PlasmaPower) [inline rai/lib/utility.cpp]: Will calling `abort()` cause problems for our test suite?

Comment (PlasmaPower) [reply rai/lib/utility.cpp]: As in, will it abort the test suite instead of just failing the test

Comment (rkeene) [reply rai/lib/utility.cpp]: This is what `assert(3)` does, so our test suite should handle it.  I did test it and this is the result:

```
[ RUN      ] block.send_serialize_json
Assertion (false) failed /home/rkeene/devel/raiblocks/rai/lib/numbers.cpp:299
Aborted
```

Comment (PlasmaPower) [reply rai/lib/utility.cpp]: Ok, if its behavior matches assert, then LGTM.

**Pull request merged**